node_modules/oae-principals/lib/rest.user.js

```diff
@@ -566,6 +567,63 @@ OAE.tenantRouter.on('post', '/api/user/:userId/picture', function(req, res) {
 });
 
 /**
+ * @REST postUserEmailVerify
+ *
+ * Verify an email token
+ *
+ * @Server      tenant
+ * @Method      POST
+ * @Path        /user/email/verify
+ * @FormParam   {string}        email                       The email to verify
+ * @FormParam   {string}        token                       The token to verify
+ * @HttpResponse                200                         The token was valid and the email address for the user that was associated with it is verified
+ * @HttpResponse                400                         A token was not specified
+ * @HttpResponse                404                         The token was not associated with a user
+ */
+OAE.tenantRouter.on('post', '/api/user/email/verify', function(req, res) {
+    PrincipalsAPI.verifyEmail(req.ctx, req.body.email, req.body.token, function(err, user) {
+        if (err) {
+            return res.send(err.code, err.msg);
+        }
+
+        // If the user is anonymous we log them into the system
+        if (!req.ctx.user()) {
+            return AuthenticationUtil.handleLogin('local', user, req, res);
```

Comment:
I think when using local authentication, there could *theoretically* be a vector of:

1. Jack updates email incorrectly, it goes to Belligerent Fool 
2. Belligerent Fool verifies email, but doesn't authenticate them, whew!
3. Belligerent Fool sends a forgot password request
4. Belligerent Fool gains access to Jack's account

Under the above 4 steps, restricting authentication on email verification isn't doing much for us.

However, under the case of SSO authentication, I think restricting authentication from email verification saves a lot of potential headache and helps me sleep at night.